xla/xla.bzl

```diff
@@ -136,5 +136,3 @@ def xla_export_hlo_deps():
 def xla_nvml_deps():
     return ["@local_config_cuda//cuda:nvml_headers"]
 
-def xla_cub_deps():
```

Comment:
You can change this if you'd like to `cub_headers` instead (whatever works is great), but I think we need something like this to help the import. I'm currently struggling to get a working patch, hopefully it will be done on Monday.

Reply:
yes, this macro is no longer needed: all CUB / HIPCUB related includes are now gathered by gpu_prim_hrds library:
https://github.com/openxla/xla/pull/6386/files#diff-35357ad3b3b70062889668b03399a71db3249cd2a01e90e0b0ed8ebe9c817272R1088-R1101
I did by analogy to TF, for me it builds & runs fine for CUDA as well as for ROCm platform.

Reply:
Hi @ddunl wonder how's this one day?

Reply:
Had a busy day today, will make sure to get something done here tomorrow. It's still a mystery why my patch behaves so poorly here. 

To add some context though about the reason why I requested that we keep `xla_cub_deps`: all macros defined in xla.bzl have a counterpart inside a file that doesn't get exported called xla.google.bzl, we use the xla.google.bzl version internally. The reason why it's necessary here is that the way we get cub debs internally is different enough from how we do it externally that regex based find and replace is difficult. So my request to keep it looks quite odd without this context

Reply:
That all being said - based on how my patch has gone it seems like I'm fundamentally misunderstanding something so it may very well be that we don't need the indirection point in xla.bzl anyway. But hopefully I can share something more positive tomorrow

Reply:
@ddunl sure, if this macro xla_cub_deps is needed for getting XLA properly integrated to TF, if I understand it correctly, then I have nothing against adding it back.